features/hooks.feature

```diff
@@ -253,6 +253,39 @@ Feature: Environment Hooks
       ]
       """
 
+  Scenario: Failing the BeforeFeatures hook exists immediately with code = 1
+    Given a file named "features/a.feature" with:
+      """
+      Feature: some feature
+
+      Scenario: I've declared one step and it is passing
+          Given This step is passing
+      """
+    And a file named "features/step_definitions/cucumber_steps.js" with:
+      """
+      var cucumberSteps = function() {
+        this.Given(/^This step is passing$/, function(callback) { callback(); });
+      };
+      module.exports = cucumberSteps;
+      """
+    And a file named "features/support/hooks.js" with:
+      """
+      var hooks = function () {
+        this.registerHandler('BeforeFeatures', function(){
+            return Promise.reject(new Error('Something bad'));
+        });
+      };
+
+      module.exports = hooks;
+      """
+    When I run cucumber.js with `-f json`
+    Then the exit status should be 1
+    And the error output contains the text: 
+    """
+      features/support/hooks.js:2 Something bad
+    """
+    And it does not output a valid JSON
```

Comment:
I'd rather we use a step like "And it output no other text". The way its written now just makes we think we are somehow printing invalid json instead of printing nothing